yt_dlp/extractor/francetv.py

```diff
@@ -358,7 +358,8 @@ def _real_extract(self, url):
             # For livestreams we need the id of the stream instead of the currently airing episode id
             video_id = traverse_obj(nextjs_data, (
                 ..., ..., 'children', ..., 'children', ..., 'children', ..., 'children', ..., ...,
-                'children', ..., ..., 'children', ..., ..., 'children', ..., 'options', 'id', {str}, any))
+                'children', ..., ..., 'children', ..., ..., 'children', (..., (..., ...)),
```

Comment:
Test URL https://www.france.tv/france-2/direct.html doesn't find a video ID with either version of this. The ID from the test-case is at `nextjs_data[1][1][3]['options']['id']`:
```suggestion
                ..., ..., (None, ('children', ..., 'children', ..., 'children', ..., 'children', ..., ...,
                'children', ..., ..., 'children', ..., ..., 'children')), (..., (..., ...)),
```

Reply:
The JSON structure seems to change every few weeks. NextJS hell

Reply:
+1
The other live test was still working, for now.

Pulled out this potential IE method FWIW, here in a _yt-dlp_ incarnation:
```py
    def _search_nextjs_data_new(self, html, video_id, paths, **kwargs):

        def find_json(s):
            return self._search_json(
                r'\w+\s*:\s*', s, 'next js data', video_id, contains_pattern=r'\[(?s:.+)\]', default=None)

        def yield_nextjs_data(html):
            for m in re.finditer(r'<script\b[^>]*>\s*self\.__next_f\.push\(\s*(\[.+?\])\s*\);?\s*</script>', html):
                yield from traverse_obj(m, (1, {json.loads}, ..., {find_json}, ...))

        return traverse_obj(yield_nextjs_data(html), (...,) + paths, **kwargs)
```